typed/union.go

```diff
@@ -105,38 +108,38 @@ type discriminator struct {
 	name string
 }
 
-func (d *discriminator) Set(m *value.Map, v discriminated) {
+func (d *discriminator) Set(m value.Map, v discriminated) {
```

Comment:
followup: why is this method public?

Reply:
It doesn't really matter since the type is private anyway. I've used it as a way to indicate that the method is public to other types within the package.

Reply:
I think I tried to do something similar once and got the feedback that it was really weird to do that. Does this pass go vet?

Reply:
Yes